File: backend/app/services/listing_media.py
"""Shared listing media defaults for imported / unclaimed centers."""

LISTING_PLACEHOLDER_IMAGE = "/images/rehab/listing-placeholder.avif"
LISTING_PLACEHOLDER_LOGO = "/images/SWA-logo-web-white-small_vSE-1.webp"
_LEGACY_PLACEHOLDERS = {
    "/images/rehab/listing-placeholder.avif",
    "/images/rehab/listing-placeholder.webp",
}


def is_placeholder_listing_image(key: str | None) -> bool:
    raw = str(key or "").strip()
    if not raw:
        return True
    return "listing-placeholder" in raw or raw in _LEGACY_PLACEHOLDERS


def is_custom_listing_image(key: str | None) -> bool:
    return not is_placeholder_listing_image(key)


def listing_image_key(key: str | None) -> str:
    return key if is_custom_listing_image(key) else LISTING_PLACEHOLDER_IMAGE

File: src/lib/listingMedia.js
export const LISTING_PLACEHOLDER_IMAGE = '/images/rehab/listing-placeholder.avif'
export const LISTING_PLACEHOLDER_LOGO = '/images/SWA-logo-web-white-small_vSE-1.webp'

export function isPlaceholderListingImage(center) {
  const src = center?.image || center?.image_url || center?.image_key || ''
  return !src || String(src).includes('listing-placeholder')
}

export function listingImageSrc(center) {
  if (isPlaceholderListingImage(center)) return LISTING_PLACEHOLDER_IMAGE
  return center?.image || LISTING_PLACEHOLDER_IMAGE
}
